.github/workflows/main.yml

```diff
@@ -267,3 +267,41 @@ jobs:
         path: ./simple_use_case/aiida/paper.pdf
         retention-days: 1
         if-no-files-found: error
+
+  run-gwl:
+    # guix uses sandboxing features which are not available in the container
+    # runtime environment. Ideally, a virtual machine with a guix installation
+    # should be used, but I have not found a way to upload a virtual machine
+    # image to github outside of using azure.
+
+    # As a workaround, a custom container image is used which already contains
+    # all software dependencies. If all dependencies are satisfied, guix doesn't
+    # use sandboxing features.
+
+    runs-on: ubuntu-latest
+    container: ghcr.io/onluen/sigwftools-gwl:latest
+
+    steps:
+      - name: checkout-repository
+        # actions/checkout v2 or higher is not supported. These version mount a
+        # node installation into the container, which seems to depend on a few
+        # system libraries. Those are not available in this image. v1 works as
+        # it doesn't use node. See
+        # https://github.com/actions/checkout/issues/334 for more details.
+        uses: actions/checkout@v1
+
+      - name: run-workflow
+        run: |
+          # Start the guix-daemon in the background, which will is queried for
+          # the workflows software environment.
+          guix-daemon > guix-daemon.log 2>&1 &
```

Comment:
What is the daemon doing? In particular when submitting that to an HPC system, is the daemon local or on the HPC?

Reply:
The daemon will only build software, it is not necessary for execution. gwl will ask the daemon to install all packages the workflow depends on. That's why it has to be running. It doesn't actually do anything, because I preseeded the image with all packages but gwl will complain, if the daemon isn't available.

Reply:
Regarding HPC: HPC nodes just need access to the packages, similar to conda. The guix-daemon also needs access to write these packages, but it doesn't matter if the daemon runs on a HPC node or on the frontend or an external node. The guix-daemon is required for some `guix` commands, but isn't required for the software installed with guix (similar to conda, unlike to docker). Unlike conda, guix provided packages are shared between all users.